app/assets/javascripts/Components/submission_file_manager.jsx

```diff
@@ -83,7 +87,8 @@ class SubmissionFileManager extends React.Component {
         delete_files: fileKeys,
         grouping_id: this.props.grouping_id
       }
-    }).then(typeof this.props.onChange === 'function' ? this.props.onChange : this.fetchData)
+    }).then(typeof this.props.onChange === 'function' ?
+        this.setState({viewFile: null, viewFileType: null, viewFileURL: null}, this.props.onChange) : this.fetchData)
```

Comment:
@ignaspa Oh sorry I just realized why this is a bit weird, the `this.setState` itself must be wrapped in a thunk (since you want to pass a callback to `then`)